hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/filtering/FilteredMessageHandler.java

```diff
@@ -44,9 +44,7 @@ public void handle(FilterResult result, Message message, Subscription subscripti
 
             updateMetrics(message, subscription);
 
-            if (subscription.isTrackingEnabled()) {
```

Comment:
I would leave this `if { ... }`, the code is synchronous and we're calling `toMessageMetadata`  on a message converter everytime, even when noOpTracker is used.